tiny_dnn/activations/softmax_layer.h

```diff
@@ -0,0 +1,49 @@
+/*
+    Copyright (c) 2017, Taiga Nomi, Karan Desai
+    All rights reserved.
+
+    Use of this source code is governed by a BSD-style license that can be found
+    in the LICENSE file.
+*/
+#pragma once
+#include "tiny_dnn/activations/activation_layer.h"
+#include "tiny_dnn/layers/layer.h"
+
+namespace tiny_dnn {
+
+class softmax_layer : public activation_layer {
+ public:
+  using activation_layer::activation_layer;
+
+  std::string layer_type() const override { return "softmax-activation"; }
+
+  void forward_activation(const vec_t &x, vec_t &y) {
+    float_t alpha = *std::max_element(x.begin(), x.end());
+    float_t denominator(0);
+    for (serial_size_t j = 0; j < x.size(); j++) {
+      y[j] = std::exp(x[j] - alpha);
+      denominator += y[j];
+    }
+    for (serial_size_t j = 0; j < x.size(); j++) {
+      y[j] /= denominator;
```

Comment:
Actually the value is `std::exp(x[i] - alpha) / denominator`, I wrote the implementation as it is right now to prevent costly calls to `std::exp` again in second for loop.